core/meta/src/TDataType.cxx

```diff
@@ -190,39 +190,39 @@ EDataType TDataType::GetType(const std::type_info &typeinfo)
 {
    EDataType retType = kOther_t;
 
-   if (!strcmp(typeid(unsigned int).name(), typeinfo.name())) {
+   if (typeid(unsigned int) == typeinfo) {
       retType = kUInt_t;
-   } else if (!strcmp(typeid(int).name(), typeinfo.name())) {
+   } else if (typeid(int) == typeinfo) {
       retType = kInt_t;
-   } else if (!strcmp(typeid(ULong_t).name(), typeinfo.name())) {
+   } else if (typeid(ULong_t) == typeinfo) {
       retType = kULong_t;
-   } else if (!strcmp(typeid(Long_t).name(), typeinfo.name())) {
+   } else if (typeid(Long_t) == typeinfo) {
       retType = kLong_t;
-   } else if (!strcmp(typeid(ULong64_t).name(), typeinfo.name())) {
+   } else if (typeid(ULong64_t) == typeinfo) {
       retType = kULong64_t;
-   } else if (!strcmp(typeid(Long64_t).name(), typeinfo.name())) {
+   } else if (typeid(Long64_t) == typeinfo) {
       retType = kLong64_t;
-   } else if (!strcmp(typeid(unsigned short).name(), typeinfo.name())) {
+   } else if (typeid(unsigned short) == typeinfo) {
       retType = kUShort_t;
-   } else if (!strcmp(typeid(short).name(), typeinfo.name())) {
+   } else if (typeid(short) == typeinfo) {
       retType = kShort_t;
-   } else if (!strcmp(typeid(unsigned char).name(), typeinfo.name())) {
+   } else if (typeid(unsigned char) == typeinfo) {
       retType = kUChar_t;
-   } else if (!strcmp(typeid(char).name(), typeinfo.name())) {
+   } else if (typeid(char) == typeinfo) {
       retType = kChar_t;
-   } else if (!strcmp(typeid(Bool_t).name(), typeinfo.name())) {
+   } else if (typeid(Bool_t) == typeinfo) {
       retType = kBool_t;
-   } else if (!strcmp(typeid(float).name(), typeinfo.name())) {
+   } else if (typeid(float) == typeinfo) {
       retType = kFloat_t;
-   } else if (!strcmp(typeid(Float16_t).name(), typeinfo.name())) {
+   } else if (typeid(Float16_t) == typeinfo) {
       retType = kFloat16_t;
-   } else if (!strcmp(typeid(double).name(), typeinfo.name())) {
+   } else if (typeid(double) == typeinfo) {
       retType = kDouble_t;
-   } else if (!strcmp(typeid(Double32_t).name(), typeinfo.name())) {
+   } else if (typeid(Double32_t) == typeinfo) {
```

Comment:
That makes no sense. But it's not your fault - it didn't make sense already before: `Double32_t` is a typedef to `double`, so this branch cannot be taken, ever. Same for `Float16_t`. OK, that's for a subsequent PR :-)